packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+// Internal Imports
+import { Component } from "../component";
+import { Events } from "../../interfaces";
+import { Tools } from "../../tools";
+import { DOMUtils } from "../../services";
+import { ChartModel } from "../../model";
+import * as Configuration from "../../configuration";
+
+// D3 Imports
+import { event, select } from "d3-selection";
+
+// import the settings for the css prefix
+import settings from "carbon-components/es/globals/js/settings";
+
+export class ToolBar extends Component {
+	type = "tool-bar";
+
+	isZoomBarEnabled: boolean;
+
+	// ul options list element
+	overflowMenuOptions;
+
+	// overflow menu icon rect to control background
+	overflowMenuIcon: any;
+
+	resetZoomMenuItemId = "resetZoomMenuItem";
+
+	// x, y coordinate of overflow menu
+	overflowMenuX = 0;
+	overflowMenuY = 0;
+
+	zoomService = this.services.zoom;
+	// overflow menu icon background padding
+	iconPadding = Configuration.toolBar.iconPadding;
+
+	constructor(model: ChartModel, services: any, configs?: any) {
+		super(model, services, configs);
+
+		this.init();
+	}
+
+	init() {
+		const options = this.model.getOptions();
+
+		// Grab the tooltip element
+		const holder = select(this.services.domUtils.getHolder());
+		const chartprefix = Tools.getProperty(options, "style", "prefix");
+
+		this.overflowMenuOptions = DOMUtils.appendOrSelect(
+			holder,
+			`div.${settings.prefix}--${chartprefix}--overflowMenu`
+		);
+
+		// Apply html content to the tooltip
+		const toolbarTextContainer = DOMUtils.appendOrSelect(
+			this.overflowMenuOptions,
+			"div.content-box"
+		);
+		this.overflowMenuOptions.style("max-width", null);
+
+		// listen to show-tooltip Custom Events to render the tooltip
+		this.services.events.addEventListener(
+			Events.Toolbar.SHOW_MENU_ITEMS,
+			() => {
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			}
+		);
+
+		// listen to hide-tooltip Custom Events to hide the tooltip
+		this.services.events.addEventListener(
+			Events.Toolbar.HIDE_MENU_ITEMS,
+			() => {
+				this.overflowMenuOptions.html(null);
+			}
+		);
+	}
+
+	render(animate = true) {
+		const isDataLoading = this.zoomService.isDataLoading();
+		this.isZoomBarEnabled =
```

Comment:
why would we want to store `isZoomBarEnabled` on `this`?

Reply:
Update in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/ba21aeb9d6dbecec4555dd58afd12bc9721fec72.